sklearn/utils/estimator_checks.py

```diff
@@ -3125,3 +3125,40 @@ def check_requires_y_none(name, estimator_orig, strict_mode=True):
 _FULLY_STRICT_CHECKS = set([
     'check_n_features_in',
 ])
+
+
+def check_transformer_get_feature_names_out(name, transformer_orig,
+                                            strict_mode=True):
+    tags = transformer_orig._get_tags()
+    if "2darray" not in tags["X_types"] or tags["no_validation"]:
+        return
+
+    X, y = make_blobs(n_samples=30, centers=[[0, 0, 0], [1, 1, 1]],
+                      random_state=0, n_features=2, cluster_std=0.1)
+    X = StandardScaler().fit_transform(X)
+    X -= X.min()
+    X = _pairwise_estimator_convert_X(X, transformer_orig)
+
+    n_samples, n_features = np.asarray(X).shape
+    transformer = clone(transformer_orig)
+    _set_checking_parameters(transformer)
+    set_random_state(transformer)
+
+    y_ = y
+    if name in CROSS_DECOMPOSITION:
+        y_ = np.c_[np.asarray(y), np.asarray(y)]
+        y_[::2, 1] *= 2
+
+    X_pred = transformer.fit_transform(X, y=y_)
+
+    input_features = ['feature%d' % i for i in range(n_features)]
+    feature_names = transformer.get_feature_names_out(input_features)
+    assert feature_names is not None
+    if isinstance(X_pred, tuple):
+        assert len(feature_names) == X_pred[0].shape[1], (
+            f"Expected {X_pred[0].shape[1]} feature names, got "
```

Comment:
It seems we don't have a test here.

Reply:
Even if we don't have an estimator for the moment, it could be good to test the check for this case.